### DISCUSSION TITLE
configdata.pm is not generated on openssl-3.0.11-1_deb12u2, if debian-amd64 option is mentioned as Configure argument

### DISCUSSION THREAD

**Discussion by debananda-pal** [openssl/openssl#23141] [Q&A] [answered]
Hi,
I found that file "configdata.pm" is not generated if I use debian-amd64 option with Configure script.

Without any modification to OpenSSL Linux source package, if I run below command, configdata.pm is not generated :
./Configure shared  --prefix=/usr --openssldir=/usr/lib/ssl --libdir=lib/x86_64-linux-gnu no-idea no-mdc2 no-rc5 no-zlib no-ssl3 enable-unit-test no-ssl3-method enable-rfc3779 enable-cms enable-ec_nistp_64_gcc_128 debian-amd64

But if I remove debian-amd64 option from above Configure command, configdata.pm file is generated.
I am using openssl-debian-openssl-3.0.11-1_deb12u2.tar.gz for the upgrade.
We were using OpenSSL 1.1.1n for buster and migrating to bookworm.

Any suggestion will be very much helpful.

Regards,
D Pal

**Comment by jamuir** [comment]
have you tried moving the platform string `debian-amd64` immediately after `./Configure` ?

**Reply by debananda-pal** [reply]
Thank you Jamuir for your reply.
I tried moving the platform string next to ./Configure, but result is same as below :

openssl-debian-openssl-3.0.11-1_deb12u2$ ./Configure debian-amd64 shared --ppenssldir=/usr/lib/ssl --libdir=lib/x86_64-linux-gnu no-idea no-mdc2 no-rc5 no-zlib no-ssl3 enable-unit-test no-ssl3-method enable-rfc3779 enablc_nistp_64_gcc_128
Configuring OpenSSL version 3.0.11 for target debian-amd64
Using os-specific seed configuration
Usage: Configure [no-<cipher> ...] [enable-<cipher> ...] [-Dxxx] [-lxxx] [-Lxxx] [-fxxx] [-Kxxx] [no-hw-xxx|no-hw] [[no-]threads] [[no-]shared] b-dynamic] [no-asm] [no-egd] [sctp] [386] [--prefix=DIR] [--openssldir=OPENSSLDIR] [--with-xxx[=vvv]] [--config=FILE] os/compiler[:flags]

pick os/compiler from:
BC-32 BS2000-OSD BSD-aarch64 BSD-generic32 BSD-generic64 BSD-ia64 BSD-riscv64
BSD-sparc64 BSD-sparcv8 BSD-x86 BSD-x86-elf BSD-x86_64 Cygwin Cygwin-i386
Cygwin-i486 Cygwin-i586 Cygwin-i686 Cygwin-x86 Cygwin-x86_64 DJGPP MPE/iX-gcc
OS390-Unix UEFI UEFI-x86 UEFI-x86_64 UWIN VC-CE VC-WIN32 VC-WIN32-ARM
VC-WIN32-ARM-UWP VC-WIN32-ONECORE VC-WIN32-UWP VC-WIN64-ARM VC-WIN64-ARM-UWP
VC-WIN64A VC-WIN64A-ONECORE VC-WIN64A-UWP VC-WIN64A-masm VC-WIN64I aix-cc
aix-gcc aix64-cc aix64-gcc aix64-gcc-as android-arm android-arm64
android-armeabi android-mips android-mips64 android-x86 android-x86_64
android64 android64-aarch64 android64-mips64 android64-x86_64 bsdi-elf-gcc cc
darwin-i386 darwin-i386-cc darwin-ppc darwin-ppc-cc darwin64-arm64
darwin64-arm64-cc darwin64-ppc darwin64-ppc-cc darwin64-x86_64
darwin64-x86_64-cc gcc haiku-x86 haiku-x86_64 hpux-ia64-cc hpux-ia64-gcc
hpux-parisc-cc hpux-parisc-gcc hpux-parisc1_1-cc hpux-parisc1_1-gcc
hpux64-ia64-cc hpux64-ia64-gcc hpux64-parisc2-cc hpux64-parisc2-gcc hurd-x86
ios-cross ios-xcrun ios64-cross ios64-xcrun iossimulator-xcrun iphoneos-cross
irix-mips3-cc irix-mips3-gcc irix64-mips4-cc irix64-mips4-gcc linux-aarch64
linux-alpha-gcc linux-aout linux-arm64ilp32 linux-armv4 linux-c64xplus
linux-elf linux-generic32 linux-generic64 linux-ia64 linux-latomic
linux-mips32 linux-mips64 linux-ppc linux-ppc64 linux-ppc64le linux-sparcv8
linux-sparcv9 linux-x32 linux-x86 linux-x86-clang linux-x86_64
linux-x86_64-clang linux32-s390x linux64-loongarch64 linux64-mips64
linux64-riscv64 linux64-s390x linux64-sparcv9 mingw mingw64 nonstop-nse
nonstop-nse_64 nonstop-nse_64_put nonstop-nse_g nonstop-nse_g_tandem
nonstop-nse_put nonstop-nse_spt nonstop-nse_spt_floss nonstop-nsv nonstop-nsx
nonstop-nsx_64 nonstop-nsx_64_put nonstop-nsx_g nonstop-nsx_g_tandem
nonstop-nsx_put nonstop-nsx_spt nonstop-nsx_spt_floss sco5-cc sco5-gcc
solaris-sparcv7-cc solaris-sparcv7-gcc solaris-sparcv8-cc solaris-sparcv8-gcc
solaris-sparcv9-cc solaris-sparcv9-gcc solaris-x86-gcc solaris64-sparcv9-cc
solaris64-sparcv9-gcc solaris64-x86_64-cc solaris64-x86_64-gcc tru64-alpha-cc
tru64-alpha-gcc uClinux-dist uClinux-dist64 unixware-2.0 unixware-2.1
unixware-7 unixware-7-gcc vms-alpha vms-alpha-p32 vms-alpha-p64 vms-ia64
vms-ia64-p32 vms-ia64-p64 vms-x86_64 vms-x86_64-cross-ia64 vos-gcc
vxworks-mips vxworks-ppc405 vxworks-ppc60x vxworks-ppc750 vxworks-ppc750-debug
vxworks-ppc860 vxworks-ppcgen vxworks-simlinux

**Comment by levitte** [comment, marked as answer]
This is a debian packaging question.  You'll probably get better answers from the package maintainers, see `debian/control` from openssl-debian-openssl-3.0.11-1_deb12u2.tar.gz.

**Reply by debananda-pal** [reply]
Can you please provide the complete link ?

**Comment by levitte** [comment]
I took another look at https://packages.debian.org/stable/openssl, and downloaded [openssl_3.0.11.orig.tar.gz](http://deb.debian.org/debian/pool/main/o/openssl/openssl_3.0.11.orig.tar.gz).  It turns out that this tarball includes everything from openssl-debian-openssl-3.0.11-1_deb12u2.tar.gz, with patches already applied.

With that one, the configuration you're trying does work as expected.

**Reply by debananda-pal** [reply]
Thenks Levitte for your reply. But I got same issue with this version as well and I don't see any debian directory in openssl_3.0.11.orig.tar.gz, which is present in openssl-debian-openssl-3.0.11-1_deb12u2.

./Configure debian-amd64 shared --prefix=/usr --openssldir=/libdir=lib/x86_64-linux-gnu no-idea no-mdc2 no-rc5 no-zlib no-ssl3 enable-unit-test no-ssl3-method enable-rfc3779 enable-cms enable-ec_nistp_64_
Configuring OpenSSL version 3.0.11 for target debian-amd64
Using os-specific seed configuration
Usage: Configure [no-<cipher> ...] [enable-<cipher> ...] [-Dxxx] [-lxxx] [-Lxxx] [-fxxx] [-Kxxx] [no-hw-xxx|no-hw] [[no-]threads] [[no-]shared] b-dynamic] [no-asm] [no-egd] [sctp] [386] [--prefix=DIR] [--openssldir=OPENSSLDIR] [--with-xxx[=vvv]] [--config=FILE] os/compiler[:flags]

pick os/compiler from:
BC-32 BS2000-OSD BSD-aarch64 BSD-generic32 BSD-generic64 BSD-ia64 BSD-riscv64
BSD-sparc64 BSD-sparcv8 BSD-x86 BSD-x86-elf BSD-x86_64 Cygwin Cygwin-i386
Cygwin-i486 Cygwin-i586 Cygwin-i686 Cygwin-x86 Cygwin-x86_64 DJGPP MPE/iX-gcc
OS390-Unix UEFI UEFI-x86 UEFI-x86_64 UWIN VC-CE VC-WIN32 VC-WIN32-ARM
VC-WIN32-ARM-UWP VC-WIN32-ONECORE VC-WIN32-UWP VC-WIN64-ARM VC-WIN64-ARM-UWP
VC-WIN64A VC-WIN64A-ONECORE VC-WIN64A-UWP VC-WIN64A-masm VC-WIN64I aix-cc
aix-gcc aix64-cc aix64-gcc aix64-gcc-as android-arm android-arm64
android-armeabi android-mips android-mips64 android-x86 android-x86_64
android64 android64-aarch64 android64-mips64 android64-x86_64 bsdi-elf-gcc cc
darwin-i386 darwin-i386-cc darwin-ppc darwin-ppc-cc darwin64-arm64
darwin64-arm64-cc darwin64-ppc darwin64-ppc-cc darwin64-x86_64
darwin64-x86_64-cc gcc haiku-x86 haiku-x86_64 hpux-ia64-cc hpux-ia64-gcc
hpux-parisc-cc hpux-parisc-gcc hpux-parisc1_1-cc hpux-parisc1_1-gcc
hpux64-ia64-cc hpux64-ia64-gcc hpux64-parisc2-cc hpux64-parisc2-gcc hurd-x86
ios-cross ios-xcrun ios64-cross ios64-xcrun iossimulator-xcrun iphoneos-cross
irix-mips3-cc irix-mips3-gcc irix64-mips4-cc irix64-mips4-gcc linux-aarch64
linux-alpha-gcc linux-aout linux-arm64ilp32 linux-armv4 linux-c64xplus
linux-elf linux-generic32 linux-generic64 linux-ia64 linux-latomic
linux-mips32 linux-mips64 linux-ppc linux-ppc64 linux-ppc64le linux-sparcv8
linux-sparcv9 linux-x32 linux-x86 linux-x86-clang linux-x86_64
linux-x86_64-clang linux32-s390x linux64-loongarch64 linux64-mips64
linux64-riscv64 linux64-s390x linux64-sparcv9 mingw mingw64 nonstop-nse
nonstop-nse_64 nonstop-nse_64_put nonstop-nse_g nonstop-nse_g_tandem
nonstop-nse_put nonstop-nse_spt nonstop-nse_spt_floss nonstop-nsv nonstop-nsx
nonstop-nsx_64 nonstop-nsx_64_put nonstop-nsx_g nonstop-nsx_g_tandem
nonstop-nsx_put nonstop-nsx_spt nonstop-nsx_spt_floss sco5-cc sco5-gcc
solaris-sparcv7-cc solaris-sparcv7-gcc solaris-sparcv8-cc solaris-sparcv8-gcc
solaris-sparcv9-cc solaris-sparcv9-gcc solaris-x86-gcc solaris64-sparcv9-cc
solaris64-sparcv9-gcc solaris64-x86_64-cc solaris64-x86_64-gcc tru64-alpha-cc
tru64-alpha-gcc uClinux-dist uClinux-dist64 unixware-2.0 unixware-2.1
unixware-7 unixware-7-gcc vms-alpha vms-alpha-p32 vms-alpha-p64 vms-ia64
vms-ia64-p32 vms-ia64-p64 vms-x86_64 vms-x86_64-cross-ia64 vos-gcc
vxworks-mips vxworks-ppc405 vxworks-ppc60x vxworks-ppc750 vxworks-ppc750-debug
vxworks-ppc860 vxworks-ppcgen vxworks-simlinux

**Reply by t8m** [reply]
@debananda-pal I am sorry but this is really a Debian packaging question. It is not something that we (OpenSSL project) ship. But maybe @kroeckx can help here.

**Reply by debananda-pal** [reply]
Thank you Tomas.

**Comment by debananda-pal** [comment]
@kroeckx, Hi Kurt,

We are trying to upgrade OpenSSL from Buster to Bookworm 
Need your help to understand below doubts.

Is the openssl-debian-openssl-3.0.11-1_deb12u2 FIPS compliant ?
a.	If the above Question answer is “No”, can I use openssl-3.0.8 ?
b.	Please let me know the changes need to add in openssl-3.0.11 (debian/control, debian/rules and other necessary files) to use openssl-3.0.8 as FIPS module.

Regards,
D Pal

**Comment by levitte** [comment]
Considering the new discussion #23154, perhaps this one should be closed?

**Reply by debananda-pal** [reply]
Sure, closing this thread.

**Comment by debananda-pal** [comment]
Discussing on #23154, so closing this.
